src/sources/fluent/mod.rs

```diff
@@ -532,15 +533,17 @@ impl TcpSourceAcker for FluentAcker {
             return None;
         }
 
-        let mut acks = String::new();
+        let mut buf = Vec::new();
+        let mut ser = Serializer::new(&mut buf);
+
         for chunk in self.chunks {
-            let ack = match ack {
-                TcpSourceAck::Ack => format!(r#"{{"ack": "{}"}}"#, chunk),
-                _ => String::from("{}"),
+            let mut ack_map = HashMap::new();
```

Comment:
I think it would be a bit more efficient to instantiate the map outside of the for loop, and clear it at the end of the loop. This preserves the allocated memory.